packages/circuit-ui/components/SidePanel/SidePanel.docs.mdx

```diff
@@ -0,0 +1,83 @@
+- **Do not** use the side panel as a permanent layout element. It should always be dismissible.
+
+## Accessibility
+
+The side panel component has been built to be used as part of a workflow and this has an effect on its accessibility.
+When the element triggering the side panel is clicked, the focus automatically shifts to the newly opened panel, facilitating a better keyboard navigation experience.
+While the side panel is visible the focus remains trapped inside. On desktop resolutions the side panel is non-modal, meaning that mouse users can easily interact with the rest of the application.
+Screen reader users would have to close the side panel first, which can be done easily using the Esc key.
+Once the side panel is closed focus is returned to the trigger element.
+
+## Usage in code
+
+First, wrap the primary content of your application in the `SidePanelProvider`.
+On desktop resolutions the primary content will be resized when the side panel is opened.
+Additionally, the `SidePanelProvider` allows for the side panels to work seamlessly with the top navigation through the `withTopNavigation` prop.
+
+```tsx
+<SidePanelProvider withTopNavigation>
+  {/* Your primary content here... */}
+</SidePanelProvider>
+```
+
+<Story id="components-sidepanel--with-top-navigation" />
+
+Then, use the `useSidePanel` hook to open a side panel from within a component.
+In the most straightforward case, simply use the `setSidePanel` function, passing the minimum set of props:
+- `children` - the content of your side panel
+- `closeButtonLabel` - an accessibility label for the close Button
+- `headline` - the title of the side panel
+
+```tsx
+import { setSidePanel, Button, Body } from '@sumup/circuit-ui';
+
+export function ShowMore() {
+  const { setSidePanel } = useSidePanel();
+
+  const handleClick = () => {
+    setSidePanel({
+      children: <Body>Some additional information</Body>,
+      closeButtonLabel: 'Close',
+      headline: 'More information',
+    });
+  };
+
+  return <Button onClick={handleClick}>Show more</Button>;
+}
+```
+
+Under the hood, `useSidePanel` generates a unique identifier for the side panel.
+When you call `setSidePanel` a second time it will close the old panel and open a new one.
+For stacked side panels you need to also provide the `backButtonLabel` accessibility prop.
+
+If you want side panels to stack using a single instance of `useSidePanel`, you'll need to pass the `group` prop with a custom identifier for each panel.
+You also need to pass `group` when you want to open and replace the same side panel from different instances of `useSidePanel`.
+
+In addition to the `setSidePanel` function the `useSidePanel` hook also returns the `updateSidePanel` and `removeSidePanel` functions.
+You can use `updateSidePanel` if you need to update the side panel asynchronously, though usually you should keep the side panel content self-contained.
```

Comment:
To clarify here: I saw the `Update and Remove` story but I'm wondering if these kind of async updates should be left to the children component's concern—or would there be a blocker here? (sorry if i'm missing something obvious)

Reply:
I added the update functionality because while I was testing the component in the Transactions list in ze-dashboard I hit a case like the one in `Update and Remove` where the side panel needs to have a loading state but the loading itself is happening in the parent component. This should be possible to avoid if the side panel content does its own data fetching, but there could be other cases where we might need to sync state between the main content and the side panel 🤷‍♂️

Reply:
Okay, this makes sense, let's keep it 👍 it's true at least for mouse users who could still interact with the rest of the app, this might come in handy. If we notice that there's little use of it, we can always decide to deprecate it later